src/datasets/iterable_dataset.py

```diff
@@ -133,6 +133,45 @@ def _shard_kwargs(shard_idx: int, kwargs: dict) -> dict:
     return {key: [value[shard_idx]] if isinstance(value, list) else value for key, value in kwargs.items()}
 
 
+def _all_shard_kwargs(kwargs: dict) -> list:
```

Comment:
Those functions' sharding logic were previously only used for iterable datasets, but now they're part of dataset builders logic, so feel free to move them in `builder.py`

Reply:
Wouldn't that cause a circular import since `builder` also imports from `iterable_dataset` and `iterable_dataset` requires `_shard_kwargs` ? Unless you mean move the other two? I'd be in favour of keeping them all in the same place

Reply:
Oh yes indeed - you can keep it this way then for now